flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ReadableCatalog.java

```diff
@@ -224,4 +224,54 @@ CatalogPartition getPartition(ObjectPath tablePath, CatalogPartitionSpec partiti
 	 */
 	boolean functionExists(ObjectPath functionPath) throws CatalogException;
 
+	// ------ Statistics ------
+
+	/**
+	 * Get the statistics of a table.
+	 *
+	 * @param tablePath path of the table
+	 * @return the statistics of the given table, null if its statistics don't exist or are unknown.
```

Comment:
Stats should probably always exist. It's how Hive handles it. Also, the behavior of "returns null when the stats doesn't exist" doesn't seem to match behaviors of other metadata APIs and would confuse people, for example `getTable()` throws `TableNotExistException` when the request table doesn't exist. It's just the default, "unknown" stats will be with some default value, for example, rowCount, fileCount, totalSize will all be 0.

This will also better explain the signatures of these APIs.

What do you think?

Reply:
Make sense. I will leave out the "null" comment in those APIs and let implementations decide. It doesn't seem making sense to throw "stats not exist" exception, though.

Reply:
Yeah, it doesn't make sense to throw `stats not exist exception`. Just using the table API and an example to demonstrate the behavior difference.